export/exporter.go

```diff
@@ -0,0 +1,61 @@
+// Package export implements a library to create
+// and define different custom OpenCensus metrics
+// exporters.
+package export
+
+import (
+	"sync"
+	"time"
+
+	"go.opencensus.io/metric/metricexport"
+)
+
+// ExporterAgent defines the wrapper format of exporters
+// and data needed by all general exporters.
+type ExporterAgent struct {
+	metricexport.Exporter
+	ir             *metricexport.IntervalReader
+	initReaderOnce sync.Once
+}
+
+// Config defines the data format of the general
+// configurations of an exporter.
+type Config struct {
+	IncludeFilter               string
+	reportingPeriodMilliseconds int
+}
+
+// NewConfig returns a new exporter Config.
+func NewConfig(filter string, reportingPeriod int) Config {
+	return Config{
+		IncludeFilter:               filter,
+		reportingPeriodMilliseconds: reportingPeriod,
+	}
+}
+
+// a user should never have to use this explicitly. They would
+// simply instantiate an implemented exporter
+func newExporterAgent(exporter metricexport.Exporter) *ExporterAgent {
+	return &ExporterAgent{
+		Exporter: exporter,
+	}
+}
+
+// Start creates the ExporterAgent's IntervalReader (if needed),
+// sets the reporting interval, and then starts the reader.
+func (e *ExporterAgent) Start(reportingPeriodms int) error {
+	e.initReaderOnce.Do(func() {
+		e.ir, _ = metricexport.NewIntervalReader(&metricexport.Reader{}, e.Exporter)
```

Comment:
The error from metricexport.NewIntervalReader() should not be ignored. It is possible e.ir would be a null pointer when the call returns error.

Reply:
I did this to copy the pattern across other OpenCensus exporter: https://github.com/census-instrumentation/opencensus-go/blob/master/examples/exporter/logexporter.go#L114-L121 and https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/cde3c106244a0c96f8b7f6027e23ff0dcf44f3fe/stats.go#L139-L144.

If e.ir returns a null pointer or something goes wrong, the error will be returned at e.ir.Start()

Reply:
do you recommend wrapping the error?

Reply:
> do you recommend wrapping the error?

The code currently does not check if e.ir is null. 
```
e.ir.ReportingInterval = time.Duration(reportingPeriodms) * time.Millisecond
```
would crash if e.ir is null. You would check for e.ir is null and return error if so